sdk/core/azure-mgmt-core/azure/mgmt/core/policies/_policy_token_header.py

```diff
@@ -0,0 +1,165 @@
+        subscription_id = None
+    if not subscription_id:
+        raise HttpResponseError("Failed to get subscriptionId from request url: {}".format(request.http_request.url))
+
+    content = getattr(request.http_request, "content", None)
+    if content and isinstance(content, str):
+        try:
+            content = json.loads(content)
+        except Exception:  # pylint: disable=broad-except
+            pass
+
+    body: Dict[str, Any] = {"operation": {"uri": request.http_request.url, "httpMethod": request.http_request.method}}
+    if content:
+        body["operation"]["content"] = content
+    change_reference = request.context.options.pop("change_reference", None)
+    if change_reference:
+        body["changeReference"] = change_reference
+
+    acquire_policy_url = "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/acquirePolicyToken"
+    return HttpRequest(
+        method="POST",
+        url=acquire_policy_url.format(subscriptionId=subscription_id),
+        params={"api-version": "2025-03-01"},
+        headers={"Accept": "application/json", "Content-Type": "application/json", "x-ms-force-sync": "true"},
+        content=json.dumps(body),
+    )
+
+
+def _update_request_with_policy_token(
+    request: PipelineRequest[HTTPRequestType],
+    acquire_policy_request: HttpRequest,
+    acquire_policy_response: Union[HttpResponse, AsyncHttpResponse],
+) -> None:
+    """Update the original request with the acquired policy token.
+
+    This method processes the response from the policy token acquisition request
+    and adds the policy token to the original request headers if successful.
+
+    :param request: The original pipeline request to update
+    :type request: ~azure.core.pipeline.PipelineRequest
+    :param acquire_policy_request: The request used to acquire the policy token
+    :type acquire_policy_request: ~azure.core.rest.HttpRequest
+    :param acquire_policy_response: The response from the policy token acquisition
+    :type acquire_policy_response: Union[~azure.core.rest.HttpResponse, ~azure.core.rest.AsyncHttpResponse]
+    :raises ~azure.core.exceptions.HttpResponseError: If policy token acquisition fails or returns non-200 status
+    """
+    if acquire_policy_response.status_code != 200:
+        raise HttpResponseError(
+            "status code is {} instead of expected 200 when trying call {} to get policy token: {}".format(
+                acquire_policy_response.status_code, acquire_policy_request.url, acquire_policy_response.text()
+            )
+        )
+
+    result = acquire_policy_response.json()
+    if result.get("result") == "Succeeded" and result.get("token"):
+        request.http_request.headers["x-ms-policy-external-evaluations"] = result["token"]
+    else:
+        raise HttpResponseError("Failed to acquire policy token: {}".format(acquire_policy_response.text()))
+
+
+class PolicyTokenHeaderPolicy(SansIOHTTPPolicy[HTTPRequestType, HTTPResponseType]):
```

Comment:
I'm not sure we need to include "header" in the policy name - as that's really just an implementation detail (e.g. `BearerTokenCredentialPolicy` and `UserAgentPolicy`). Maybe something like `ExternalPolicyEvaluationPolicy`?
Or `PolicyEvaluationTokenPolicy`?

As an aside, as this implementation contains IO, I don't believe this should derive from `SansIOHTTPPolicy`.

Reply:
(1) Updated code to make it derive from HTTPPolicy same as async one.
(2) I don't have much preference about the naming. `PolicyTokenHeaderPolicy` / `ExternalPolicyEvaluationPolicy` / `PolicyEvaluationTokenPolicy`, which one do you like? @xiangyan99

Reply:
Let's adhere to our architects' naming guidelines. :)